### ISSUE TITLE
Tool not returning correct output to the agent when it contains a Wait node

### ISSUE THREAD

**Issue by lamarof** [n8n-io/n8n#12414]
### Bug Description

When the tool contains a Wait node things get messed up

### To Reproduce

insert a Wait node in the tool and run the workflow of the agent. the agent will not produce an output in the json return.

### Expected behavior

when the tool does not contain a Wait node, the agent outputs what it did correctly. otherwise it does not provide an output unless you look at the logs, where you can see the output it should produce.

### Operating System

Windows

### n8n Version

latest n8n cloud 

### Node.js Version

latest n8n cloud

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @lamarof,

We have created an internal ticket to look into this which we will be tracking as "N8N-8027"

**Comment by lamarof** [comment]
Hi,

Exactly, I added a wait node in a workflow tool.

Best,
Francesco

**Comment by jeanpaul** [comment]
@lamarof Hey, can I ask you to share your workflow JSONs then? I wasn't able to reproduce this locally with just a wait-node in my sub-workflow.

**Comment by lamarof** [comment]
Hey,

I was just making some trials and it happened, so I thought It would be
helpful to tell you.

Did you make sure that the sub-workflow was called by the agent in the main
workflow?

Best,
Francesco

**Comment by jeanpaul** [comment]
Yes, the sub-workflow was called. Thanks for your report. If it happens more often, please let us know. We'll also keep an eye out. For now, I'll close this issue.

**Comment by lamarof** [comment]
Thank you jeanpaul. have a nice weekend.

francesco

**Comment by blambert** [comment]
Hello all, I have some more troubleshooting information on this issue.

I have a workflow that has started doing this.  I was working on a self-hosted Docker instance three releases behind 1.81.4.  Once it started happening, I updated to 1.81.4 and it's still doing it. This is not related to an "agent," but just general usage.

In my case, when viewing the main workflow, the output it shows for the sub-workflow is the **input of the sub-workflow** (the output of the immediately preceding node).  When I click "View sub-execution" and observe the output of the last node in the sub-workflow, it shows the expected output.

In my case, the last node in the sub-workflow is a Code node that creates the one field that makes up the desired output.  It does contain a Wait node (on webhook). The sub-workflow works perfectly.

I have pinned and unpinned data. I have exported the sub-workflow as JSON and recreated it.  I have exported the "super-workflow" as JSON and recreated it.

The only thing that makes it work is changing the Wait node to "After time interval."

It appears to me that a Wait on Webhook node causes the sub-workflow to pass the output of the trigger ("When executed by another workflow") to the output of the workflow.
